libraries/Microsoft.Bot.Builder/BotFrameworkAdapter.cs

```diff
@@ -932,5 +939,25 @@ private async Task<MicrosoftAppCredentials> GetAppCredentialsAsync(string appId,
             _appCredentialMap[appId] = appCredentials;
             return appCredentials;
         }
+
+        /// <summary>
+        /// Middleware to assign tenantId from channelData to Conversation.TenantId.
+        /// </summary>
+        internal class TenantIdWorkaroundForTeamsMiddleware : IMiddleware
+        {
```

Comment:
Please add a far more detailed comment here. Why is this needed? When can we deprecate it?

Reply:
Added additional documentation to describe why and when can this be removed.